authorization/resource/resource.go

```diff
@@ -0,0 +1,223 @@
+	List(ctx context.Context) ([]Resource, error)
+	IsValid(context.Context, string) bool
+	Search(ctx context.Context, q string, start int, limit int) ([]Resource, int, error)
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m *GormResourceRepository) TableName() string {
+	return "resource"
+}
+
+// CRUD Functions
+
+// Load returns a single Resource as a Database Model
+func (m *GormResourceRepository) Load(ctx context.Context, id string) (*Resource, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "resource", "load"}, time.Now())
+
+	var native Resource
+	err := m.db.Table(m.TableName()).Where("id = ?", id).Find(&native).Error
+	if err == gorm.ErrRecordNotFound {
+		return nil, errs.WithStack(errors.NewNotFoundError("resource", id))
+	}
+
+	return &native, errs.WithStack(err)
+}
+
+// CheckExists returns nil if the given ID exists otherwise returns an error
+func (m *GormResourceRepository) CheckExists(ctx context.Context, id string) error {
+	defer goa.MeasureSince([]string{"goa", "db", "resource", "exists"}, time.Now())
+	return repository.CheckExists(ctx, m.db, m.TableName(), id)
+}
+
+// Create creates a new record.
+func (m *GormResourceRepository) Create(ctx context.Context, model *Resource) error {
+	defer goa.MeasureSince([]string{"goa", "db", "resource", "create"}, time.Now())
+
+	// If no identifier has been specified for the new resource, then generate one
+	if model.ID == "" {
+		model.ID = uuid.NewV4().String()
+	}
+	err := m.db.Create(model).Error
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"resource_id": model.ID,
+			"err":         err,
+		}, "unable to create the resource")
+		return errs.WithStack(err)
+	}
+	log.Info(ctx, map[string]interface{}{
+		"resource_id": model.ID,
+	}, "Resource created!")
+	return nil
+}
+
+// Lookup looks for an existing resource with the given ID
+func (m *GormResourceRepository) Lookup(ctx context.Context, id string) (*Resource, error) {
+	if id == "" {
+		return nil, errs.New("Cannot lookup resource with empty resource ID")
+	}
+	log.Debug(nil, nil, "Looking for resource with id=%s\n", id)
+	// bind the assignee to an existing resource
+	resource, err := m.First(ResourceFilterByID(id))
+	if err != nil {
+		return nil, errs.Wrapf(err, "failed to lookup resource by id '%s'", id)
+	}
+	// use existing resource
+	log.Debug(nil, nil, "Using existing resource with ID: %v", resource.ID)
+	return resource, nil
+}
+
+// Save modifies a single record.
+func (m *GormResourceRepository) Save(ctx context.Context, model *Resource) error {
+	defer goa.MeasureSince([]string{"goa", "db", "resource", "save"}, time.Now())
+
+	obj, err := m.Load(ctx, model.ID)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"resource_id": model.ID,
+			"ctx":         ctx,
+			"err":         err,
+		}, "unable to update the resource")
+		return errs.WithStack(err)
+	}
+	err = m.db.Model(obj).Updates(model).Error
```

Comment:
I think using m.db.Save(&user) is safer than using Updates() in our Save() methods.
Save() updates all fields but Updates() updates only changed and not blank(!). So if resource.Description is an empty string then it wont be updated!
See more details here: http://jinzhu.me/gorm/crud.html#update

Overall our space resource repo https://github.com/fabric8-services/fabric8-auth/blob/master/space/space_resource.go is probably a better example of CRUD methods.